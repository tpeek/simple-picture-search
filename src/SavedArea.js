import React from 'react';
import Link from '@material-ui/core/Link';
import { makeStyles } from '@material-ui/core/styles';
import OpenInNew from '@material-ui/icons/OpenInNew';

const useStyles = makeStyles({
  root: {
    float: 'right',
    borderLeft: 'solid lightgray 2px',
    height: 700,
    paddingLeft: 40,
    minWidth: 150,
  },
  link: {
    fontWeight: 'bold',
    display: 'block',
    padding: '4px 0',
    margin: '10px 0',
    fontVariantNumeric: 'tabular-nums',
  },
  openIcon: {
    marginLeft: 10,
    marginBottom: -2,
    width: 14,
    height: 14,
  },
});

const SavedArea = ({ savedResults }) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <h2>Saved</h2>
      {savedResults.map((result) => (
        <Link key={result.id} target="_blank" href={result.pageURL} className={classes.link}>
          #{result.id}
          <OpenInNew className={classes.openIcon} />
        </Link>
      ))}
    </div>
  );
};

export default SavedArea;
