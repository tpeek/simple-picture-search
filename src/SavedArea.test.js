import React from 'react';
import SavedArea from './SavedArea';

const defaultProps = {
  savedResults: [{ id: 1, pageURL: 'https://www.example.com' }],
};

const setup = (props) => shallow(<SavedArea {...defaultProps} {...props} />);

describe('<SavedArea />', () => {
  it('renders snapshot correctly', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('renders snapshot correctly when there are no savedResults', () => {
    const wrapper = setup({ savedResults: [] });

    expect(wrapper).toMatchSnapshot();
  });
});
