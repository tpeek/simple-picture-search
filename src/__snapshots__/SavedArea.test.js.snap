// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SavedArea /> renders snapshot correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <h2>
    Saved
  </h2>
  <WithStyles(ForwardRef(Link))
    className="makeStyles-link-2"
    href="https://www.example.com"
    key="1"
    target="_blank"
  >
    #
    1
    <OpenInNewIcon
      className="makeStyles-openIcon-3"
    />
  </WithStyles(ForwardRef(Link))>
</div>
`;

exports[`<SavedArea /> renders snapshot correctly when there are no savedResults 1`] = `
<div
  className="makeStyles-root-1"
>
  <h2>
    Saved
  </h2>
</div>
`;
